skimage/filters/thresholding.py

```diff
@@ -818,27 +819,40 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            :DOI:`10.1117/12.767755`
     """
-    if w == 1 or w % 2 == 0:
-        raise ValueError(
-            "Window size w = %s must be odd and greater than 1." % w)
+    def _sanity_check_window_size(axis_size):
+        if axis_size == 1 or axis_size % 2 == 0:
+            raise ValueError(
+                "Window size w = {} must be odd and greater than 1."
```

Comment:
Why does it need to be greater than 1?

Reply:
I am confuse, what would be effect of kernel of size 1 ? Doesn't it indicate that mean will be the pixel value it self and std 0 ?

Reply:
the case ``w=1`` is trivial (unchanged image for mean, zero standard deviation)

Reply:
Guys, the whole point of this PR is to allow rectangular window sizes. A window size of (1, 1, 1) indeed makes no sense, but (1, 50, 50) might make sense for an image stack.

Reply:
Or it could just be ignored? If someone gives w=1, a whole bunch of computation will be wasted, but nothing more dramatic will happen.

Reply:
I agree with @jni to ignore it.